Description: update Makefile
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2019-11-14
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,16 @@
-CC = gcc
-CXX = g++
-CFLAGS = -O2 -ffast-math -fomit-frame-pointer -Irubber1.0
-CXXFLAGS = -O2 -ffast-math -fomit-frame-pointer -Irubber1.0 -std=c++0x
-LIBS = /usr/lib/$(shell dpkg-architecture -qDEB_HOST_MULTIARCH)/libbass.so -lglut -lGLU -lGL -lm
+# compilers and tools
+SHELL = /bin/sh
+CC := gcc
+CXX := g++
+PKG_CONFIG := pkg-config
+RM_F       := rm -f
+
+# Flags
+CFLAGS += -Wall -pipe -Irubber1.0 $(shell $(PKG_CONFIG) --cflags glu)
+CXXFLAGS += -Wall -pipe -Irubber1.0 -std=c++11 $(shell $(PKG_CONFIG) --cflags glu)
+
+# Libraries for linking
+LDFLAGS += /usr/lib/$(shell dpkg-architecture -qDEB_HOST_MULTIARCH)/libbass.so -lglut $(shell $(PKG_CONFIG) --libs glu) -lm
 
 OBJ = test3d.o Kewlmation.o \
       rubber1.0/Rubber.o rubber1.0/Shader.o rubber1.0/Texture.o \
@@ -10,9 +18,42 @@
       rubber1.0/Fbo.o rubber1.0/BassRocket.o \
       rocket/sync/data.o rocket/sync/device.o rocket/sync/track.o
 
-onefinger: $(OBJ)
-	$(CXX) -o $@ $(OBJ) $(LIBS)
-	strip $@
+# Final executable name
+TARGET=1finger-by-kewlers
+
+# Install variables
+INSTALL := install
+INSTALL_DIR     := $(INSTALL) -p -d -m  755
+INSTALL_PROGRAM := $(INSTALL) -p    -m  755
+INSTALL_DATA    := $(INSTALL) -p    -m  644
+INSTALL_DATAR   := cp -r
+
+prefix          := /usr
+exec_prefix     := $(prefix)
+bindir          := $(prefix)/bin
+datarootdir     := $(prefix)/share
+datadir         := $(prefix)/share/1finger-by-kewlers
+
+
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
+
+all: $(TARGET)
+
+$(TARGET): $(OBJ)
+	$(CXX) -o $@ $(OBJ) $(LDFLAGS)
 
 clean:
-	-rm $(OBJ) onefinger
+	-@$(RM_F) $(OBJ) $(TARGET)
+
+install: $(TARGET)
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)
+	$(RM_F) $(DESTDIR)$(bindir)/$(TARGET)
+	$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)$(bindir)
+	#data
+	$(INSTALL_DIR) $(DESTDIR)$(datadir)
+	$(INSTALL_DATAR) data $(DESTDIR)$(datadir)
